api/types/authentication_test.go

```diff
@@ -64,3 +68,144 @@ func TestEncodeDecodeRequireMFAType(t *testing.T) {
 		})
 	}
 }
+
+func TestAuthPreference_CheckAndSetDefaults(t *testing.T) {
+	for _, tt := range []struct {
+		name           string
+		spec           AuthPreferenceSpecV2
+		assertErr      require.ErrorAssertionFunc
+		assertAuthPref func(t *testing.T, authPref AuthPreference)
+	}{
+		{
+			name: "OK default to OTP",
+			spec: AuthPreferenceSpecV2{},
+			assertAuthPref: func(t *testing.T, authPref AuthPreference) {
+				require.Equal(t, []SecondFactorType{SecondFactorType_SECOND_FACTOR_TYPE_OTP}, authPref.GetSecondFactors())
+			},
+		},
+		{
+			name: "OK OTP",
+			spec: AuthPreferenceSpecV2{
+				SecondFactors: []SecondFactorType{
+					SecondFactorType_SECOND_FACTOR_TYPE_OTP,
+				},
+			},
+			assertAuthPref: func(t *testing.T, authPref AuthPreference) {
+				require.False(t, authPref.GetAllowPasswordless())
```

Comment:
1. Assert GetSecondFactors / GetSecondFactor directly? I think we care more about the flag than the derived settings. Same for others.

2. nit: Use `assert` instead of `require`?

Reply:
I don't really care about `SecondFactor` here and `GetSecondFactors` just returns the list we set above. My intention with these tests was to test these default settings like passwordless. I'll add a separate test for converting SecondFactor to SecondFactors.